### PR TITLE
Add boosted trees estimators

### PR DESCRIPTION
**Pull request by kevinykuo**
This change
- Adds `boosted_trees_classifier()` and `boosted_trees_regressor()`
- Skips coercion of numpy arrays in `input_fn()`. Prior to this PR, `as.array()` is called, which casts numeric variables to `float64`.

Closes https://github.com/rstudio/tfestimators/issues/146

Note that integer labels are not supported for classification in TensorFlow. Will add tests when that's resolved. Tracking at https://github.com/rstudio/tfestimators/issues/151.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# tfestimators 1.5.9001
+
+* `input_fn.list()` no longer performs type coercion on numpy arrays.
+
+* Added `boosted_trees_regressors()` and `boosted_trees_classifier()` (#146).
+
+* Added a `NEWS.md` file to track changes to the package.
```
Comment (terrytangyuan) [inline NEWS.md]: I think you can remove this line since it's pretty obvious

**Comment by kevinykuo** [comment]
@terrytangyuan Thanks! Nope haven't tried on larger datasets yet.

**Comment by terrytangyuan** [comment]
Just realized that we should also update API reference and README/index.Rmd.

**Comment by kevinykuo** [comment]
OK will track at https://github.com/rstudio/tfestimators/issues/153

**Pull request merged**